tests/unit/utils/test_auth_dependency.py

```diff
@@ -96,3 +129,35 @@ def test_auth_dependency_config(_setup):
     assert isinstance(
         authz_client, AuthorizationV1Api
     ), "authz_client is not an instance of AuthorizationV1Api"
+
+
+@patch("ols.utils.auth_dependency.K8sClientSingleton.get_custom_objects_api")
+def test_get_cluster_id(mock_get_custom_objects_api):
+    """Test get_cluster_id function."""
+    cluster_id = {"spec": {"clusterID": "some-cluster-id"}}
+    mocked_call = MagicMock()
+    mocked_call.get_cluster_custom_object.return_value = cluster_id
+    mock_get_custom_objects_api.return_value = mocked_call
+    assert get_cluster_id() == "some-cluster-id"
```

Comment:
prob. use real UUID? will it be tested in the code btw?

Reply:
Do you mean ensuring what we get from the cluster as id is really uuid? That seems like an overkill to me, but I can add it if you think it is reasonable.